cpp/tests/test_abm.cpp

```diff
@@ -238,63 +246,201 @@ TEST(TestLocation, beginStep)
-    params.get<mio::abm::CriticalToRecovered>()[{age, vaccination_state}] = 0.5;
-    params.set<mio::abm::RecoveredToSusceptible>({{mio::abm::AgeGroup::Count, mio::abm::VaccinationState::Count}, 0.});
-    params.get<mio::abm::RecoveredToSusceptible>()[{age, vaccination_state}] = 0.5;
-    params.set<mio::abm::SusceptibleToExposedByCarrier>(
-        {{mio::abm::AgeGroup::Count, mio::abm::VaccinationState::Count}, 0.});
-    params.get<mio::abm::SusceptibleToExposedByCarrier>()[{age, vaccination_state}] = 0.4;
-    params.set<mio::abm::SusceptibleToExposedByInfected>(
-        {{mio::abm::AgeGroup::Count, mio::abm::VaccinationState::Count}, 0.});
-    params.get<mio::abm::SusceptibleToExposedByInfected>()[{age, vaccination_state}] = 0.5;
 
-    //setup location with some chance of exposure
-    auto home      = mio::abm::Location(mio::abm::LocationType::Home, 0, 0);
-    auto location1 = mio::abm::Location(mio::abm::LocationType::PublicTransport, 0, 3);
-    auto infected1 = mio::abm::Person(home, mio::abm::InfectionState::Carrier, mio::abm::AgeGroup::Age15to34, params,
-                                      vaccination_state);
+    auto home = mio::abm::Location(mio::abm::LocationType::Home, 0);
+    home.set_capacity(4, 264);
+    auto location = mio::abm::Location(mio::abm::LocationType::PublicTransport, 0);
+    location.set_capacity(4, 264);
+
+    auto infected1 = mio::abm::Person(home, mio::abm::InfectionState::Carrier, age, params, vaccination_state);
     home.add_person(infected1);
-    infected1.migrate_to(home, location1, {0});
-    auto infected2 = mio::abm::Person(home, mio::abm::InfectionState::Infected, mio::abm::AgeGroup::Age80plus, params,
-                                      vaccination_state);
-    home.add_person(infected2);
-    infected2.migrate_to(home, location1, {0, 1});
-    auto infected3 = mio::abm::Person(home, mio::abm::InfectionState::Infected, mio::abm::AgeGroup::Age5to14, params,
-                                      vaccination_state);
-    home.add_person(infected3);
-    infected3.migrate_to(home, location1, {1});
+    auto infected2 = mio::abm::Person(home, mio::abm::InfectionState::Carrier, age, params, vaccination_state);
+    location.add_person(infected2);
 
-    //cache precomputed results
-    auto dt = mio::abm::seconds(8640);
-    location1.begin_step(dt, params);
+    location.capacity_adapted_transmission_risk_flag();
+
+    home.compute_relative_transmission_risk();
+    location.compute_relative_transmission_risk();
 
-    ASSERT_TRUE(std::abs(location1.get_cells()[0].cached_exposure_rate[{age, vaccination_state}] - 0.9) < 0.001);
-    ASSERT_TRUE(std::abs(location1.get_cells()[1].cached_exposure_rate[{age, vaccination_state}] - 1) < 0.001);
-    ASSERT_TRUE(std::abs(location1.get_cells()[2].cached_exposure_rate[{age, vaccination_state}]) < 0.001);
+    ASSERT_EQ(location.compute_relative_transmission_risk(), 0.25);
+    ASSERT_EQ(home.compute_relative_transmission_risk(), 1.0);
```

Comment:
For double values please use an ASSERT which has a tolerance of the order of machine precision.

Reply:
You can use EXPECT_NEAR(a, b, tol)

tol = 1e-14 should do it. If that's to small, go with 1e-12